src/components/Overview/PlotlyCharts/ForceDirectedGraph.tsx

```diff
@@ -1,14 +1,174 @@
+    });
+    // Set the initial links (You, Facebook, Google)
+    links.push({
+      source: 'You',
+      target: 'Facebook',
+      value: 10,
+    });
+    links.push({
+      source: 'You',
+      target: 'Google',
+      value: 10,
+    });
+
+    // Build nodes and links based on each type of relationships with companies identified in files
+    for (let i = 0; i < data.length; i++) {
+      if (
+        // @ts-ignore
+        data[i].action_type ===
+        'Advertisers whose ads you have recently seen or clicked on'
+      ) {
+        nodes.push({
+          id: data[i].details,
+          group: 4,
+        });
+        links.push({
+          source: 'You',
+          target: data[i].details,
+          value: 3,
+        });
+        links.push({
+          source: data[i].details,
+          target: data[i].platform,
+          value: 3,
+        });
+      } else if (
+        // @ts-ignore
+        data[i].action_type ===
+        'Advertiser who has added your name to an audience based on your information or your activity outside of Facebook'
+      ) {
+        nodes.push({
+          id: data[i].details,
+          group: 4,
+        });
+        links.push({
+          source: 'You',
+          target: data[i].details,
+          value: 3,
+        });
+        links.push({
+          source: data[i].details,
+          target: data[i].platform,
+          value: 3,
+        });
+      } else if (
+        // @ts-ignore
+        data[i].action_type ===
+        'Company that shared information with facebook about your activity on their website/application'
+      ) {
+        nodes.push({
+          id: data[i].details,
+          group: 5,
+        });
+        links.push({
+          source: 'You',
+          target: data[i].details,
+          value: 3,
+        });
+        links.push({
+          source: data[i].details,
+          target: data[i].platform,
+          value: 3,
+        });
+      } else if (
+        // @ts-ignore
+        data[i].action_type ===
+        'Apps that have been installed on your mobile device'
+      ) {
+        nodes.push({
+          id: data[i].details,
+          group: 6,
+        });
+        links.push({
+          source: 'You',
+          target: data[i].details,
+          value: 3,
+        });
+        links.push({
+          source: data[i].details,
+          target: data[i].platform,
+          value: 3,
+        });
+      } else if (
+        // @ts-ignore
+        data[i].action_type === 'Account linked to your Facebook profile'
+      ) {
+        nodes.push({
+          id: data[i].details,
+          group: 7,
+        });
+        links.push({
+          source: 'You',
+          target: data[i].details,
+          value: 3,
+        });
+        links.push({
+          source: data[i].details,
+          target: data[i].platform,
+          value: 3,
+        });
+        // @ts-ignore
+      } else if (data[i].action_type === 'Game you played on Facebook') {
```

Comment:
Beaucoup de `else if`, ne serait-il pas mieux d'avoir une interface dans dataTypes.ts avec tout ce que tu souhaites parser ? et tu itères pour chaque datapoint à travers cette liste, ce qui tu permet d'avoir qu'une condition if au lieu de toutes celles présentes ici... + non scalable si on ajoute une multitude de plateformes.
A dispo pour en discuter.

Reply:
Commentaire outdated.